### PR TITLE
[TH2-3785-5.0v] verification refactored and necessary mocks added

### PR DESCRIPTION
**Pull request by nChurgulia**

### CODE REVIEW

File: src/main/java/com/exactpro/th2/mstore/AbstractMessageStore.java [hunk 1]
```diff
@@ -287,24 +279,46 @@ private static String formatMessageBatchToStore(GroupedMessageBatchToStore batch
      */
     private void verifyBatch(T delivery) {
         List<M> messages = getMessages(delivery);
+        HashMap<SessionKey, SessionData> innerCache = new HashMap<>();
```
Comment (georgiano) [inline src/main/java/com/exactpro/th2/mstore/AbstractMessageStore.java:282]: when is the innerCache applied to global cache?

File: src/main/java/com/exactpro/th2/mstore/AbstractMessageStore.java [hunk 2]
```diff
@@ -287,24 +279,46 @@ private static String formatMessageBatchToStore(GroupedMessageBatchToStore batch
...
+                prevSequence = sessionData.get(sessionKey).lastSequence.get();
+            } else {
+                prevSequence = getLastMessageSequence(sessionKey);
+            }
+
             if (lastKey == null) {
                 lastKey = sessionKey;
             } else {
-                verifySession(i, lastKey, sessionKey);
+                verifySessionGroup(i, lastKey, sessionKey);
```
Comment (georgiano) [inline src/main/java/com/exactpro/th2/mstore/AbstractMessageStore.java:301]: is it ever checked that messages of same session across batches have same group?

File: src/main/java/com/exactpro/th2/mstore/AbstractMessageStore.java [hunk 3]
```diff
@@ -316,11 +330,11 @@ private static void verifySequence(int messageIndex, long lastSeq, long currentS
         }
     }
 
-    private static void verifySession(int messageIndex, SessionKey lastKey, SessionKey sessionKey) {
-        if (!lastKey.equals(sessionKey)) {
+    private static void verifySessionGroup(int messageIndex, SessionKey lastKey, SessionKey sessionKey) {
```
Comment (georgiano) [inline src/main/java/com/exactpro/th2/mstore/AbstractMessageStore.java:333]: it makes sense to discard this function and move it's logic in caller function

File: src/test/java/com/exactpro/th2/mstore/TestCaseMessageStore.java [hunk 4]
```diff
@@ -77,6 +77,15 @@ void setUp() throws CradleStorageException, IOException {
         when(storageMock.getEntitiesFactory()).thenReturn(cradleEntitiesFactory);
         when(storageMock.storeGroupedMessageBatchAsync(any(GroupedMessageBatchToStore.class))).thenReturn(completableFuture);
 
+        StoredMessage mockedStoredMessage = mock(StoredMessage.class);
+        when(mockedStoredMessage.getTimestamp()).thenReturn(Instant.MIN);
```
Comment (georgiano) [inline src/test/java/com/exactpro/th2/mstore/TestCaseMessageStore.java:81]: why are you degrading getTimestamp() function?

**Review by georgiano — CHANGES_REQUESTED** [review]

File: src/main/java/com/exactpro/th2/mstore/AbstractMessageStore.java (hunk 1 above)
Comment (nChurgulia) [reply src/main/java/com/exactpro/th2/mstore/AbstractMessageStore.java:282]: in the handle method after calling verifyBatch all messages are added to global cache

File: src/main/java/com/exactpro/th2/mstore/AbstractMessageStore.java (hunk 2 above)
Comment (nChurgulia) [reply src/main/java/com/exactpro/th2/mstore/AbstractMessageStore.java:301]: in verifySessionGroup method groups of lastKey and sessionKey are compared and it throws exception if they are different

File: src/main/java/com/exactpro/th2/mstore/AbstractMessageStore.java (hunk 3 above)
Comment (nChurgulia) [reply src/main/java/com/exactpro/th2/mstore/AbstractMessageStore.java:333]: Ok

**Pull request merged**